### PR TITLE
Ensure keyring is for current user

### PR DESCRIPTION
**Pull request by patrickxb**

### CODE REVIEW

File: go/libkb/account.go [hunk 1]
```diff
@@ -281,16 +281,18 @@ func (a *Account) Keyring() (*SKBKeyringFile, error) {
...
 	// not sure how this could happen, but just in case:
 	if unp == nil {
 		return nil, NoUsernameError{}
 	}
+
+	if a.skbKeyring != nil && a.skbKeyring.IsForUsername(*unp) {
+		a.G().Log.Debug("Account: found loaded keyring for %s", *unp)
+		return a.skbKeyring, nil
+	}
+
```
Comment (maxtaco) [inline go/libkb/account.go]: set `a.skbKeyring = nil` here in case we error out below?

File: go/engine/login_provision.go [hunk 2]
```diff
@@ -852,12 +860,24 @@ func (e *loginProvision) makeEldestDevice(ctx *Context) error {
...
+	aerr := e.G().LoginState().Account(func(a *libkb.Account) {
+		a.EnsureUsername(e.arg.User.GetNormalizedName())
+		ctx.LoginContext = a
+
+		if err = e.makeDeviceKeys(ctx, args); err != nil {
+			return
+		}
+
+		// save provisioned device id in the session
+		err = a.LocalSession().SetDeviceProvisioned(e.G().Env.GetDeviceID())
```
Comment (maxtaco) [inline go/engine/login_provision.go]: can you use my new `GetDeviceIDForUsername` function here?

File: go/libkb/account.go (hunk 1 above)
Comment (patrickxb) [inline go/libkb/account.go]: sure

File: go/engine/login_provision.go (hunk 2 above)
Comment (patrickxb) [inline go/engine/login_provision.go]: done

**Pull request merged**